### PR TITLE
Unit test git recursive submodule

### PR DESCRIPTION
**Pull request by joaobrittoneto**
Should fix https://github.com/rock-core/autobuild/issues/94

### CODE REVIEW

**Review by g-arjones — APPROVED** [review]

File: lib/autobuild/import/git.rb [hunk 1]
```diff
@@ -1201,7 +1201,7 @@ def checkout(package, _options = Hash.new)
             update_remotes_configuration(package)
             update(package, only_local: !remote_branch.start_with?("refs/"),
                             reset: :force)
-            run_git(package, "submodule", "update", '--init') if with_submodules?
+            run_git(package, "submodule", "update", '--init', '--recursive') if with_submodules?
```
Comment (g-arjones) [inline lib/autobuild/import/git.rb:1204]: Actually, as I mentioned in the issue I don't think you need `--recursive` here

Comment (joaobrittoneto) [reply lib/autobuild/import/git.rb]: The tests fails when the `--recursive` is removed from the checkout method.
It seems that there is a problem with the method with `--recurse-submodules`.

```
Finished in 36.352230s, 10.0957 runs/s, 12.1863 assertions/s.

  1) Failure:
Autobuild::Git::nested submodules handling::checkout#test_0003_checkouts the submodule of the local branch [/home/runner/work/autobuild/autobuild/test/import/test_git.rb:727]:
--- expected
+++ actual
@@ -1,2 +1,2 @@
-"Commit 0
+"Commit 1
 "


  2) Failure:
Autobuild::Git::nested submodules handling::checkout#test_0002_checkouts submodules at the state of the tag/commit pin [/home/runner/work/autobuild/autobuild/test/import/test_git.rb:722]:
--- expected
+++ actual
@@ -1,2 +1,2 @@
-"Commit 0
+"Commit 1
 "
```

Comment (joaobrittoneto) [reply lib/autobuild/import/git.rb]: Other possibility is to replace this [update](https://github.com/rock-core/autobuild/blob/master/lib/autobuild/import/git.rb#L1247)
```
update(package, only_local: !remote_branch.start_with?("refs/"),
                            reset: :force)
```
with
```
update(package, only_local: false, reset: :force)
```
The tests pass, but I'm not entirely sure why ` only_local: !remote_branch.start_with?("refs/")` was used here

Comment (joaobrittoneto) [reply lib/autobuild/import/git.rb]: @doudou any suggestions here?
It seems to have some problem with the checkout method with the `--recurse-submodule`. Not sure the best place to fix it though

Comment (doudou) [reply lib/autobuild/import/git.rb]: Did some blame-ing. The `only_local` argument to `update` in `checkout` is to optimize in the very common case that you're checking out a branch that is already available locally (a.k.a. a "normal branch"). If remote branch is a full ref, we assume it's a "non-head" branch, that is for instance a pull request, that needs to be fetched explicitly.

Before we added support for these types of refs, it was always `false`, which is the reason why there is a need for a `submodule init` just afterwards (and to add `--recursive` to it).

We *could* do the `submodule init` call only in cases where only_local is true, but I'm not sure it's worth the optimization (submodules being rare in general).

**Comment by doudou** [comment]
Hey. The tests still fail. As I mentioned in my last comment, I believe just keeping the `--recurse` in place would be good for me.

**Comment by joaobrittoneto** [comment]
Sorry, I lost track of it. The `--recursive` is back. Tests should pass now.

**Comment by doudou** [comment]
> Sorry, I lost track of it. The --recursive is back. Tests should pass now.

Just having some style issues.

**Review by doudou — APPROVED** [review]

**Pull request merged**